src/azure-cli/azure/cli/command_modules/config/custom.py

```diff
@@ -36,6 +44,12 @@ def config_set(cmd, key_value=None, local=False):
                 name = parts[1]
 
                 cmd.cli_ctx.config.set_value(section, name, _normalize_config_value(value))
+                if section in telemetry_allowed_sections_or_keys or key in telemetry_allowed_sections_or_keys:
```

Comment:
I would suggest to put this filtering logic into telemetry module to make `config_set` as simple as possible.

Reply:
As this extra `debug_info` is command specific and in the future we may add other data for other modules. Putting it in telemetry module may need extra handling of each command module and less flexible.